infer/environment.cc

```diff
@@ -1010,6 +1010,21 @@ core::TypePtr Environment::processBinding(core::Context ctx, const cfg::CFG &inW
                         }
                     }
 
+                    if (it->main.method.exists() && it->main.method.data(ctx)->flags.isPackagePrivate) {
+                        core::ClassOrModuleRef klass = core::Types::getClassForAppliedOrClassType(it->main.receiver);
```

Comment:
This implementation is buggy. Here's a failing test case showing why:

https://github.com/sorbet/sorbet/compare/jez-failing-package-private-test#diff-5b422b38a3eb48152c151254cbda3be8a2aff7878eed8228597112bd23fe3712R20-R21

This code assumes that `it->main.receiver` will always be a class type or applied type, but that's not true. It could be many types.

What you really want to look at is `it->main.method.data(ctx)->owner`, which will always exist and always be a `ClassOrModuleRef`.

In general, there's a reason why no method like `getClassForAppliedOrClassType` exists in the codebase: it's a huge crutch encouraging people to conflate types with symbols. If you could also delete it while fixing this, that would be greatly appreciated!